partner_contact_block_invoice_address/models/res_partner.py

```diff
@@ -0,0 +1,71 @@
+# Copyright 2024 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
+
+
+from odoo import _, api, models
+from odoo.exceptions import UserError
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    def _check_setting_block_invoice_address(self):
+        partner_block_invoice_address = (
+            self.env["ir.config_parameter"]
+            .sudo()
+            .get_param("partner_block_invoice_address")
+        )
+        if partner_block_invoice_address == "True":
+            partner_block_invoice_address_default_type = (
+                self.env["ir.config_parameter"]
+                .sudo()
+                .get_param("partner_block_invoice_address_default_type")
+            )
+            return partner_block_invoice_address_default_type
+        return False
+
+    @api.onchange("type")
+    def _onchange_contact_type(self):
+        default_type = self._check_setting_block_invoice_address()
+        if default_type and self.type == "invoice":
+            self.type = default_type
```

Comment:
I would first check is `self.type == "invoice"` then call `_check_setting_block_invoice_address`